src/Build/BackEnd/Components/RequestBuilder/IntrinsicTasks/ItemGroupIntrinsicTask.cs

```diff
@@ -429,11 +429,16 @@ ISet<string> removeMetadata
                     try
                     {
                         // Pass the non wildcard expanded excludes here to fix https://github.com/Microsoft/msbuild/issues/2621
-                        string[] includeSplitFiles = EngineFileUtilities.GetFileListEscaped(
+                        (string[] includeSplitFiles, FileMatcher.SearchAction action) = EngineFileUtilities.GetFileListEscaped(
                             Project.Directory,
                             includeSplit,
                             excludes);
 
+                        if (action == FileMatcher.SearchAction.ReturnLogDriveEnumerationWildcard)
+                        {
+                            LoggingContext.LogWarning($"Drive enumeration was detected during wildcard expansion, for the Include attribute, which relied on the file spec {includeSplit}.");
```

Comment:
![image](https://user-images.githubusercontent.com/89041446/144299224-fc979fe8-67f1-4432-89a3-f65830bb3e63.png)

I'm not certain as to why this would cause an internal error. Does `LoggingContext` require a resource name?

Reply:
Output with new changes --
![image](https://user-images.githubusercontent.com/89041446/144509290-f057b831-8fb3-4884-b7d3-da3c02544291.png)